jcl/src/java.base/share/classes/java/lang/reflect/Array.java

```diff
@@ -71,7 +71,7 @@ public static Object get(Object array, int index) throws IllegalArgumentExceptio
 		return ((short[])array)[index];
 	} else if (arrayClass == byte[].class) {
 		/* Avoiding Byte cache yields 5x performance improvement. */
-		return new Byte(((byte[])array)[index]);
+		return Byte.valueOf(((byte[])array)[index]);
```

Comment:
Hopefully this comment is no longer true: Perhaps it should be deleted?

Reply:
This was an optimization that was made when we switched from the native implementation to the Java implementation as the Java implementation was more "JIT friendly". Perhaps @andrewcraik knows if this is still the case.

Reply:
If nothing else, it should be changed to match the pattern of the rest of the code.

Reply:
I do not know if this is still the case - that conversion happened quite a long time ago. The code would need to be benchmarked to know if the change to Byte.valueOf is performance neutral and if bypassing the Byte cache still yields a performance boost. I think any decision to change this needs perf testing since it was clearly important and so we need to verify the best way to handle the operations. Doing the new directly could allow for stack allocation when valueof may not (due to the cache pathways). I am not sure why the optimization was only applied in this specific case, but it being different is not sufficient to warrant a change without perf evaluation IMO.